src/App.js

```diff
@@ -1,23 +1,58 @@
 import React, { Component } from 'react';
-import Main from './Main.js'
+
+import Main from './Main.js';
+import Login from './components/Login';
+import Logout from './components/Logout';
+import api from './api';
+
 import 'bootstrap/dist/css/bootstrap.min.css';
 import './App.css';
 
 class App extends Component {
-
   constructor(props) {
     super(props);
     this.state = {
-      user: null
-    }
+      user: null,
+      users: [],
+    };
   }
 
-  render() {
+  async componentDidMount() {
+    const users = await api.getAllUsers();
+    this.setState({ users });
+  }
 
+  onLogin = async (profileObj) => {
+    let users = this.state.users;
+    let user = users.find((u) => u.email === profileObj.email);
+    if (!user) {
+      await api.createUser({
+        name: profileObj.givenName,
+        email: profileObj.email,
+        private: true,
+      });
+
+      users = await api.getAllUsers();
+    }
+    this.setState({ user, users });
+  };
+
+  onLogout = () => {
+    this.setState({ user: null });
+  };
+
+  render() {
     return (
-      <div className="App">
-        <img id="logo" src={require('./s13_logo.jpg')} alt='logo'/>
-        <Main user={this.state.user}/>
+      <div className='App'>
```

Comment:
I know this is old, but why the capital letter in the class name?

Reply:
It's the React base, I didn't write it lol